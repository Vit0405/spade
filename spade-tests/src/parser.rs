use crate::snapshot_error;

snapshot_error! {
    stage_outside_pipeline,
    "
    entity main(x: X) -> int<8> {
        reg;
    }
    "
}

snapshot_error! {
    register_count_is_required,
    "
    pipeline(3) main(x: X) -> int<8> {
        reg *;
    }
    "
}
