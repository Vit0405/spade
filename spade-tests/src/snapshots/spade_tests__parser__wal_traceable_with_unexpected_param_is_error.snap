---
source: spade-tests/src/parser.rs
---
#[wal_traceable(a, uses_clk, this_is_not_valid)]
struct T {}


error: Invalid parameter for wal_traceable
  ┌─ testinput:1:17
  │
1 │ #[wal_traceable(a, uses_clk, this_is_not_valid)]
  │                 ^ Invalid parameter
  │
  = note: wal_traceable only takes the parameters suffix, uses_clk or uses_rst


